### PR TITLE
Clarify that LightningModule requires forward to be implemented

### LINKED ISSUE #21472: LightningModule -Inference in Research- phrasing issue
### 📚 Documentation

In lightning_module.rst (https://github.com/Lightning-AI/pytorch-lightning/blob/master/docs/source-pytorch/common/lightning_module.rst#inference-in-research), the sentence:

“you can add a forward method to the LightningModule”

is misleading because LightningModule requires forward to be implemented.

cc @lantiga

### PR DESCRIPTION
**Pull request by CodeVishal-17**
## What does this PR do?

This PR fixes a misleading sentence in the LightningModule documentation under
"Inference in Research".

The previous wording suggested that a `forward` method can be optionally added,
while in practice `LightningModule` requires `forward` to be implemented.
The change clarifies this requirement without altering any behavior.

Fixes #21472

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
  - Yes, tracked in #21472
- [x] Did you read the contributor guideline, **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**?
- Did you make sure to **update the documentation** with your changes?
  - Yes
- Did you write any **new necessary tests**?
  - Not applicable (documentation-only change)
- [ ] Did you verify new and **existing tests pass** locally?
  - Not applicable (documentation-only change)
- Did you list all the **breaking changes** introduced by this pull request?
  - No breaking changes
- Did you **update the CHANGELOG**?
  - Not required for documentation-only changes

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21476.org.readthedocs.build/en/21476/

### CODE REVIEW

**Comment by bhimrazy** [comment]
My bad 😅 — I mistakenly added a changelog commit here that was meant for another PR. I’ve reverted it and restored the original commit now.

**Review by tchaton — COMMENTED** [review]
The forward method isn't strickly required but the training_step, validation_step, test_step. You can add a forward method to reduce the duplicated code.

**Comment by CodeVishal-17** [comment]
@tchaton Thanks for the clarification — that makes sense 👍  

You’re right: `forward` itself isn’t strictly required as long as `training_step` / `validation_step` / `test_step` are implemented. My intent with this PR was to fix the misleading wording that suggested *no execution path at all* was required.

I’m happy to update the wording to explicitly state that:
- a LightningModule must implement at least one execution entry point (`training_step`, etc.), and
- `forward` is optional but recommended to reduce duplication.

Let me know if that wording sounds right and I’ll push a small update.

**Review by deependujha — DISMISSED** [review]
Hi @CodeVishal-17, thanks for iterating on this. I’m not convinced we need to update the wording here.

`forward` is optional in `LightningModule`. Inference can be done entirely via `predict_step`, and `forward` is primarily useful for research-style or manual inference when calling the module directly.

This is already demonstrated in the existing documentation:

* `predict_step`-only inference examples
* and the note that *the advantage of adding a forward* is enabling more involved inference logic (e.g. text generation)

So the sentence “you can add a forward method” is intentional and accurate, and the surrounding sections already clarify when and why you would do so. Updating it to imply a requirement would be misleading.

```python
# no training_step or any such method required
class LightningTransformer(L.LightningModule):
    def __init__(self, vocab_size):
        super().__init__()
        self.model = Transformer(vocab_size=vocab_size)

    def forward(self, batch):
        inputs, target = batch
        return self.model(inputs, target)

    def configure_optimizers(self):
        return torch.optim.SGD(self.model.parameters(), lr=0.1)


model = LightningTransformer(vocab_size=dataset.vocab_size)

model.eval()
with torch.no_grad():
    batch = dataloader.dataset[0]
    pred = model(batch)
```

cc: @bhimrazy

**Comment by CodeVishal-17** [comment]
Thanks everyone for the clarifications and detailed discussion.

I understand now that `forward` is intentionally optional in `LightningModule`,
and inference can be handled entirely via `predict_step`.
Given that, updating the wording here could indeed be misleading.

I’ll close this PR to avoid introducing confusion.
Thanks again for taking the time to explain the design intent.
